### PR TITLE
Load

### LINKED ISSUE #4: Add a `load` syntax form to the annotation language
We need to add a new term, `load`, to the annotation language to allow us to model load (and eventually store) instructions explicitly. 

For now, the first steps are to:
1. Add `load` to the parser.
2. Add a new corresponding `veri_ir::Expr` enum variant. 

Later steps will be to actually have the solver reason over equivalence of loads, but we will save that for a later issue.

### PR DESCRIPTION
**Pull request by aw125888**
add a load syntax form to the annotation language. Fixes #4

### CODE REVIEW

File: cranelift/isle/veri/veri_engine/src/annotations.rs [hunk 1]
```diff
@@ -259,6 +259,19 @@ fn spec_op_to_expr(s: &SpecOp, args: &Vec<SpecExpr>, pos: &Pos, env: &ParsingEnv
                 .collect();
             Expr::Switch(Box::new(swith_on), arms)
         }
+        SpecOp::Load => {
+            assert_eq!(
+                args.len(),
+                3,
+                "Unexpected number of args for subs operator {:?}",
```
Comment (avanhatt) [inline cranelift/isle/veri/veri_engine/src/annotations.rs]: Typo fix: This error message should say `load`, not `subs`

**Review by avanhatt — CHANGES_REQUESTED** [review]
Mostly good, but a few small changes needed:
1. Typo fix as shown in the comment. 
2. A test is failing: I will look into fixing this for you.

File: cranelift/isle/veri/veri_engine/src/annotations.rs (hunk 1 above)
Comment (aw125888) [reply cranelift/isle/veri/veri_engine/src/annotations.rs]: We changed the message to include load instead of subs.

**Review by avanhatt — APPROVED** [review]
Looks good once the checks pass

**Pull request merged**